stdlib/toolchain/CompatibilitySpan/CMakeLists.txt

```
@@ -2,6 +2,13 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND DEFINED SWIFT_STDLIB_LIBRARY_BUILD_TY
 
   set(library_name "swiftCompatibilitySpan")
 
+  # The correct fix would be to add this to the following to the invocation of add_swift_target_library
+  # DEPLOYMENT_VERSION_MACCATALYST ${COMPATIBILITY_MINIMUM_DEPLOYMENT_VERSION_MACCATALYST}
+  # but the underlying logic takes the deployment version from the cache instead
+  # so for now we override those values locally
+  set(SWIFT_DARWIN_DEPLOYMENT_VERSION_OSX ${COMPATIBILITY_MINIMUM_DEPLOYMENT_VERSION_OSX})
+  set(SWIFT_DARWIN_DEPLOYMENT_VERSION_MACCATALYST ${COMPATIBILITY_MINIMUM_DEPLOYMENT_VERSION_MACCATALYST})
```

I would set these deployment versions explicitly, especially the macOS one which is higher that the 10.9 we use for the other back deployment archives.

```suggestion
  set(SWIFT_DARWIN_DEPLOYMENT_VERSION_OSX "10.14.4")
  set(SWIFT_DARWIN_DEPLOYMENT_VERSION_MACCATALYST "13.1")
```